python/cudf_polars/cudf_polars/utils/config.py

```diff
@@ -318,21 +316,61 @@ class StreamingExecutor:
     with the 'distributed' scheduler.
     """
 
+    _env_prefix = "CUDF_POLARS__EXECUTOR"
+
     name: Literal["streaming"] = dataclasses.field(default="streaming", init=False)
-    scheduler: Scheduler = Scheduler.SYNCHRONOUS
+    scheduler: Scheduler = dataclasses.field(
+        default_factory=_make_default_factory(
+            f"{_env_prefix}__SCHEDULER", Scheduler.__call__, Scheduler.SYNCHRONOUS
+        )
+    )
     fallback_mode: StreamingFallbackMode = dataclasses.field(
-        default_factory=streaming_fallback_mode_default
+        default_factory=_make_default_factory(
+            f"{_env_prefix}__FALLBACK_MODE",
+            StreamingFallbackMode.__call__,
+            StreamingFallbackMode.WARN,
+        )
+    )
+    max_rows_per_partition: int = dataclasses.field(
+        default_factory=_make_default_factory(
+            f"{_env_prefix}__MAX_ROWS_PER_PARTITION", int, 1_000_000
+        )
+    )
+    unique_fraction: dict[str, float] = dataclasses.field(
+        default_factory=_make_default_factory(
+            f"{_env_prefix}__UNIQUE_FRACTION", json.loads, {}
+        )
     )
-    max_rows_per_partition: int = 1_000_000
-    unique_fraction: dict[str, float] = dataclasses.field(default_factory=dict)
     target_partition_size: int = dataclasses.field(
-        default_factory=target_partition_size_default
+        default_factory=_make_default_factory(
+            f"{_env_prefix}__TARGET_PARTITION_SIZE", int, 0
+        )
+    )
+    groupby_n_ary: int = dataclasses.field(
+        default_factory=_make_default_factory(f"{_env_prefix}__GROUPBY_N_ARY", int, 32)
+    )
+    broadcast_join_limit: int = dataclasses.field(
+        default_factory=_make_default_factory(
+            f"{_env_prefix}__BROADCAST_JOIN_LIMIT", int, 0
+        )
+    )
+    shuffle_method: ShuffleMethod = dataclasses.field(
+        default_factory=_make_default_factory(
+            f"{_env_prefix}__SHUFFLE_METHOD",
+            ShuffleMethod.__call__,
+            ShuffleMethod.TASKS,
+        )
+    )
+    rapidsmpf_spill: bool = dataclasses.field(
+        default_factory=_make_default_factory(
+            f"{_env_prefix}__RAPIDSMPF_SPILL", _bool_converter, default=False
+        )
+    )
+    sink_to_directory: bool | None = dataclasses.field(
+        default_factory=_make_default_factory(
+            f"{_env_prefix}__SINK_TO_DIRECTORY", _bool_converter, None
```

Comment:
See my earlier note/question about whether the `None` default needs to be reflected in the `_make_default_factory` type-hinting.

Reply:
I *think* that things are correct, but let me check:

- We let the default `sink_to_directory=None`, which means "True `scheduler=distributed`, False for `scheduler="synchronous"`. No change from `branch-25.08` here
- *If* you configure `sink_to_directory` through an environment variable, it *must* resolve to a boolean. In other words, `CUDF_POLARS__EXECUTOR__SINK_TO_DIRETORY=0` is allowed (and will be checked later to ensure it's compatible with the scheduler), but `CUDF_POLARS__EXECUTOR__SINK_TO_DIRETORY=none` is not allowed. I think this is OK, since AFAICT there's no reason to set that (it's already the default).

So assuming we're OK with not allowing `CUDF_POLARS__EXECUTOR__SINK_TO_DIRETORY=none` then we should be good.

Reply:
Just to clarify, the functionality definitely seems correct to me.

I'm just noting that the `Callable[[], T]` return type of `_make_default_factory` may not be entirely "correct" in this case, because `T` is `bool`, but the callable may return `None` if no environment variable is set?

Feel free to ignore me if I'm just confused :)

Reply:
Ah, I see. But somehow mypy infers the type of `T` here to be `bool | None`. Adding:

```python
reveal_type(_make_default_factory("foo", _bool_converter, default=None))
```

shows

```
python/cudf_polars/cudf_polars/utils/config.py:564: note: Revealed type is "def () -> Union[builtins.bool, None]"
```

That's actually want in this case, but I don't see how it's type safe. I guess that `bool | None` is the only type that would work here... You've gotten me confused too :)